tests/unit/blocking_connection_tests.py

```diff
@@ -204,3 +208,33 @@ def test_sleep(self, select_connection_class_mock):
             connection.sleep(0.00001)
 
 
+class _SimpleProxy(object):
+
+    def __init__(self, obj):
+        self._obj = obj
+
+    def __getattribute__(self, item):
+        if item in ('_obj', 'send'):
+            return object.__getattribute__(self, item)
+        return getattr(self._obj, item)
+
+
+class TestChannelCreationOnBrokenConnection(unittest.TestCase):
+
+    def test_that_it_raise_closed_connection(self):
+
+        for error in (errno.EPIPE, errno.ECONNABORTED, errno.ECONNREFUSED):
+            # initiate a connection:
+            conn = blocking_connection.BlockingConnection()
```

Comment:
yeah, that was the easier solution I found actually.
I agree with most of your comments/remarks but otherwise I would have to handle a full socket(.socket) mock which was harder to me. I just found a lib to do that, giving it a look right now.. (https://github.com/mocketize/python-mocket).. would it makes the job so here ?